src/vendor/components/navigation/index.jsx

```diff
@@ -0,0 +1,132 @@
+// eslint-disable react/prop-types
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import isEqual from 'lodash/isEqual';
+import { frum, zipObject } from '../../queryOps';
+import { Object2URL, URL2Object } from '../../convert';
+
+function withNavigation(config) {
+  // https://reactjs.org/docs/higher-order-components.html
+  //
+  // Deals with synchronizing url parameters with this.state, and
+  // selector widget states, and url history
+  //
+  // Expects a `config` that is a list of configurations for various
+  // selection widgets. Right now there is only Picker.
+  //
+  // Adds properties to `props`:
+  // * `navigation` - that contains a Navigation component
+  //   to include on `render()`
+  // * properties with names frum(config).select("id")
+
+  return WrappedComponent => {
```

Comment:
I think we'll want to return the class otherwise we would be returning nothing:

```jsx
// no braces at the end
return WrappedComponent =>
  class Output extends React.Component {
```

Reply:
We `return withRouter(withStyles(styles)(Output));`  which is need to provide `history` and styling to the navigation component.